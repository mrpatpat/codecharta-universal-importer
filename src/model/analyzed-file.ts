export class AnalyzedFile {
    
    private metrics: Metrics;

    constructor(private path: string) {
        this.metrics = {};
    }

    public addMetrics(metrics: Metrics) {
        Object.assign(this.metrics, metrics);
    }

    public getPath(): string {
        return this.path;
    }

    public getMetrics(): Metrics {
        return this.metrics;
    }

}