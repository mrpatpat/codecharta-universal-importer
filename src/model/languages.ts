export enum Languages {
    TYPESCRIPT,
    JAVASCRIPT,
    ALL,
    UNKNOWN
}