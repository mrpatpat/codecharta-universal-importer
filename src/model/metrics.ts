interface Metrics {
    [key: string]: string;
}