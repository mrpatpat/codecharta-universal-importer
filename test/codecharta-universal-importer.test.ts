import CodechartaUniversalImporter from "../src/codecharta-universal-importer"
import { AnalyzedFile } from "../src/model/analyzed-file";

describe("CodechartaUniversalImporter", () => {
  it("works if true is truthy", () => {
    expect(true).toBeTruthy()
  })

  it("CodechartaUniversalImporter is instantiable", () => {
    expect(new CodechartaUniversalImporter()).toBeInstanceOf(CodechartaUniversalImporter)
  })

  it("should return correct analyzed files", () => {
    const i = new CodechartaUniversalImporter();
    const analyzed: AnalyzedFile[] = i.analyzeDirectory("test/sample-projects/typescript");
    expect(analyzed).toMatchSnapshot();
  })

  it("should correctly analyze prettierJS", () => {
    const i = new CodechartaUniversalImporter();
    const analyzed: AnalyzedFile[] = i.analyzeDirectory("test/sample-projects/javascript/prettier");
    expect(analyzed).toMatchSnapshot();
  })
  
  it("should correctly analyze files and export into a cc map", () => {
    const i = new CodechartaUniversalImporter();
    const cc: string = i.buildCodechartaFromDirectory("test/sample-projects/javascript/prettier", "PrettierJS");
    expect(cc).toMatchSnapshot();
  })

})
