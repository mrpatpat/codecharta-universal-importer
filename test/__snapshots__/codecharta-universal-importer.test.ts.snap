// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodechartaUniversalImporter should return correct analyzed files 1`] = `
Array [
  AnalyzedFile {
    "metrics": Object {
      "sloc_block_lines": "0",
      "sloc_comment_lines": "0",
      "sloc_empty_lines": "0",
      "sloc_mixed_lines": "0",
      "sloc_single_lines": "0",
      "sloc_source_lines": "1",
      "sloc_todo_lines": "0",
      "sloc_total_lines": "1",
    },
    "path": "test/sample-projects/typescript/hello-old-world.js",
  },
  AnalyzedFile {
    "metrics": Object {
      "sloc_block_lines": "0",
      "sloc_comment_lines": "0",
      "sloc_empty_lines": "3",
      "sloc_mixed_lines": "0",
      "sloc_single_lines": "0",
      "sloc_source_lines": "8",
      "sloc_todo_lines": "0",
      "sloc_total_lines": "11",
      "sonarts_classes": "1",
      "sonarts_cognitiveComplexity": "0",
      "sonarts_commentLines": "0",
      "sonarts_complexity": "2",
      "sonarts_executableLines": "2",
      "sonarts_functions": "2",
      "sonarts_ncloc": "8",
      "sonarts_nosonarLines": "0",
      "sonarts_statements": "2",
    },
    "path": "test/sample-projects/typescript/hello-universe.ts",
  },
  AnalyzedFile {
    "metrics": Object {
      "sloc_block_lines": "0",
      "sloc_comment_lines": "0",
      "sloc_empty_lines": "3",
      "sloc_mixed_lines": "0",
      "sloc_single_lines": "0",
      "sloc_source_lines": "8",
      "sloc_todo_lines": "0",
      "sloc_total_lines": "11",
      "sonarts_classes": "1",
      "sonarts_cognitiveComplexity": "0",
      "sonarts_commentLines": "0",
      "sonarts_complexity": "2",
      "sonarts_executableLines": "2",
      "sonarts_functions": "2",
      "sonarts_ncloc": "8",
      "sonarts_nosonarLines": "0",
      "sonarts_statements": "2",
    },
    "path": "test/sample-projects/typescript/hello-world.ts",
  },
]
`;
